### PR TITLE
In AMQP broker, when trying to reconnect, after connection was

### LINKED ISSUE #411: Publisher doesn't reconnect to broker (rabbit mq)
Setup:
We setup machinery using version 1.6.2 with result backend redis and broker RabbitMQ. We run 5 worker with concurrency set to 10. We have 1 publisher (our web process which is embedded in iris framework). We test if publisher reconnects to RabbitMQ in case of dropping connection (we turn down MQ process for some time and turn it back on after some time) 

Actual:
Worker processes are able to reconnect to RabbitMQ. Publisher process is not able to reconnect (even after long time), reusing previously created publisher server process results into error `Could not send task: %sSet state pending error: Dial error: Exception (501) Reason: "EOF"`

Expected:
Worker and publisher should reconnect to RabbitMQ, in case connection was propped for some time.

### PR DESCRIPTION
**Pull request by ronniel1**
disconnected, remove the connection for the queue, so that
GetOrOpenConnection() will retry the connection.
Fix #411

### CODE REVIEW

File: v1/brokers/amqp/amqp.go [hunk 1]
```diff
@@ -232,7 +235,7 @@ func (b *Broker) Publish(ctx context.Context, signature *tasks.Signature) error
 			DeliveryMode: amqp.Persistent,
 		},
 	); err != nil {
-		return err
+		return errors.Wrap(err, "Failed to get publish task")
```
Comment (michaelf-stratoscale) [inline v1/brokers/amqp/amqp.go]:
```suggestion
		return errors.Wrap(err, "Failed to publish task")
```

File: v1/brokers/amqp/amqp.go [hunk 2]
```diff
@@ -214,7 +217,7 @@ func (b *Broker) Publish(ctx context.Context, signature *tasks.Signature) error
 		amqp.Table(b.GetConfig().AMQP.QueueBindingArgs), // queue binding args
 	)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "Failed to get a connection")
```
Comment (michaelf-stratoscale) [inline v1/brokers/amqp/amqp.go]: consider adding `queue` to the error reply

**Comment by ilyamochalov** [comment]
@ronniel1 Hi there. I appreciate you started the work the bug I reported. Can we move forward with fixing it? I really the issue to be fixed... Can I help anyhow?

File: v1/brokers/amqp/amqp.go (hunk 2 above)
Comment (ronnie-stratoscale) [reply v1/brokers/amqp/amqp.go]: done

File: v1/brokers/amqp/amqp.go (hunk 1 above)
Comment (ronnie-stratoscale) [reply v1/brokers/amqp/amqp.go]: done

**Comment by ronnie-stratoscale** [comment]
@ilyamochalov I guess we just need to wait for @RichardKnop  to merge this

**Pull request merged**